mergify_engine/actions/assign.py

```diff
@@ -28,19 +28,23 @@
 
 
 class AssignAction(actions.Action):
-    validator = {
-        # NOTE: "users" is deprecated, but kept as legacy code for old config
-        voluptuous.Required("users", default=[]): [types.Jinja2],
-        voluptuous.Required("add_users", default=[]): [types.Jinja2],
-        voluptuous.Required("remove_users", default=[]): [types.Jinja2],
-    }
-
     flags = (
         actions.ActionFlag.ALLOW_AS_ACTION
         | actions.ActionFlag.ALLOW_ON_CONFIGURATION_CHANGED
         | actions.ActionFlag.ALWAYS_RUN
     )
 
+    @classmethod
+    def get_config_schema(
+        cls, partial_validation: bool
+    ) -> typing.Dict[typing.Any, typing.Any]:
+        return {
+            # NOTE: "users" is deprecated, but kept as legacy code for old config
+            voluptuous.Required("users", default=[]): [types.Jinja2],
```

Comment:
It's not obvious, but you can actually use `default=list` and voluptuous will see it's a callable and call it.

That might be a simpler change?

Reply:
https://github.com/Mergifyio/mergify-engine/pull/3150